taskcat/_cfn/threaded.py

```diff
@@ -0,0 +1,236 @@
+import logging
+import uuid
+from functools import partial
+from multiprocessing.dummy import Pool as ThreadPool
+from typing import Dict, List
+
+import boto3
+
+from taskcat._cfn.stack import Parameter, Stack, Stacks, StackStatus, Tag
+from taskcat._common_utils import merge_dicts
+from taskcat._config import Config
+from taskcat._config_types import AWSRegionObject, Test
+from taskcat.exceptions import TaskCatException
+
+LOG = logging.getLogger(__name__)
+
+
+def fan_out(func, partial_kwargs, payload, threads):
+    pool = ThreadPool(threads)
+    if partial_kwargs:
+        func = partial(func, **partial_kwargs)
+    results = pool.map(func, payload)
+    pool.close()
+    pool.join()
+    return results
+
+
+class Stacker:
+
+    NULL_UUID = uuid.UUID(int=0)
+
+    def __init__(
+        self,
+        config: Config,
+        uid: uuid.UUID = NULL_UUID,
+        stack_name_prefix: str = "tCaT",
+        tags: list = None,
+    ):
+        self.config = config
+        self.project_name = config.name
+        self.stack_name_prefix = stack_name_prefix
+        self.tags = tags if tags else []
+        self.uid = uuid.uuid4() if uid == Stacker.NULL_UUID else uid
+        self.stacks: Stacks = Stacks()
+
+    @staticmethod
+    def _tests_to_list(tests: Dict[str, Test]):
+        return [test for test in tests.values()]
+
+    def create_stacks(self, threads: int = 8):
+        if self.stacks:
+            raise TaskCatException("Stacker already initialised with stack objects")
+        tests = self._tests_to_list(self.config.tests)
+        tags = [Tag({"Key": "taskcat-id", "Value": self.uid.hex})]
+        tags += [
+            Tag(t)
+            for t in self.tags
+            if t.key not in ["taskcat-project-name", "taskcat-test-name", "taskcat-id"]
+        ]
+        fan_out(self._create_stacks_for_test, {"tags": tags}, tests, threads)
+
+    def _create_stacks_for_test(self, test, tags, threads: int = 32):
+        stack_name = "{}-{}-{}-{}".format(
+            self.stack_name_prefix, self.project_name, test.name, self.uid.hex
```

Comment:
Suggestion:
```
name_section_list = []
for name_section in [self.stack_name_prefix, self.project_name, test.name, self.uid.hex]:
  if name_section:
     name_section_list.append(name_section)
stack_name = name_section_list.join('-')
```

This would automagically handle `self.stack_name_prefix == None`